staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -251,6 +252,15 @@ type Interface interface {
 	// TODO: Remove when storage.Interface will be separate from etc3.store.
 	// Deprecated: Added temporarily to simplify exposing RequestProgress for watch cache.
 	RequestWatchProgress(ctx context.Context) error
+
+	// Supports checks whether the etcd instance supports the specified features and returns true if it does,
+	// or false otherwise. It utilizes the provided context.Context for cancellation or timeout control.
+	// The function accepts one or more featuregate.Feature parameters representing the features to check.
+	//
+	// TODO(ah8ad3): Consider enhancing the feature checking mechanism by potentially combining
```

Comment:
Let's remove this TODO - let's not optimize for something we're not yet sure if it will really provide some value.